pkg/network/macvlan/macvlan.go

```diff
@@ -87,8 +87,14 @@ func Create(name string, master string, netns ns.NetNS) (*netlink.Macvlan, error
 }
 
 // Install configures a macvlan interfaces created with Create method
-func Install(link *netlink.Macvlan, ips []*net.IPNet, routes []*netlink.Route, netns ns.NetNS) error {
+func Install(link *netlink.Macvlan, hw net.HardwareAddr, ips []*net.IPNet, routes []*netlink.Route, netns ns.NetNS) error {
 	return netns.Do(func(_ ns.NetNS) error {
+		if hw != nil && len(hw) != 0 {
```

Comment:
`len(hw) != 0` ? what iz dees ?

Reply:
Because HardwareAddr is just an alias to a byte slice. So this check makes sure that not only that the array is allocated, but that it also has elementes (bytes) in it. Idd as u suggest, a len(hw) != 6 is a better check.